### PR TITLE
feat: portal tooltips

### PR DESCRIPTION
**Pull request by corteggiano**
## Overview
portaling out tooltips so they dont mess with iot-app-kit/dashboard z-index

tooltip

Screenshot 2023-04-11 at 10 19 18
Screenshot 2023-04-11 at 10 19 13

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx [hunk 1]
```diff
@@ -55,6 +83,116 @@ export class ScTooltip {
     this.dataContainer.removeEventListener('mousedown', this.hideTooltip);
   }
 
+  tooltipHeight = (numRows: number) => numRows * TOOLTIP_ROW_HEIGHT + TOOLTIP_EMPTY_HEIGHT;
+
+  visualizedDataStreams = (): DataStream[] => {
```
Comment (corteggiano) [inline packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx:88]: most changes here were functions i had to move over from `sc-tooltip-rows.tsx` to get positioning working :)

File: packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+import { activePoints, POINT_TYPE } from '../sc-webgl-base-chart/activePoints';
+import { isDefined, isSupportedDataType } from '../../../utils/predicates';
+import { TooltipPoint, TooltipPositioning } from './types';
+import { StreamType } from '../../../utils/dataConstants';
+import { trendLinePoints } from './trendLinePoints';
+import { sortTooltipPoints } from './sort';
+
+const TOOLTIP_ROW_HEIGHT = 21;
+const TOOLTIP_EMPTY_HEIGHT = 71;
+const X_OFFSET = 16;
```
Comment (tracy-french) [inline packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx:17]: Nit: Magic numbers. Please fix.

File: packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx [hunk 3]
```diff
@@ -68,6 +206,11 @@ export class ScTooltip {
       const selectedDateMS = start.getTime() + viewportDuration * ratio;
 
       this.selectedDate = new Date(selectedDateMS);
+
+      this.positionPortal();
+
+      const tooltipContent = this.el.querySelector('iot-app-kit-vis-tooltip-rows') as HTMLElement | undefined;
```
Comment (tracy-french) [inline packages/iot-app-kit-visualizations/src/components/charts/sc-tooltip/sc-tooltip.tsx:212]: Love that you cast as `Maybe`.

**Review by tracy-french — APPROVED** [review]
Looks great. Please fix the magic numbers when you have the capacity.

**Pull request merged**